pkgs/development/python-modules/pyrad/default.nix

```diff
@@ -28,33 +41,34 @@ buildPythonPackage rec {
     })
   ];
 
-  nativeBuildInputs = [
-    poetry-core
-  ];
+  nativeBuildInputs = [ poetry-core ];
 
   propagatedBuildInputs = [
-    netaddr
+    netaddr_0_8_0
     six
   ];
-
   preCheck = ''
     substituteInPlace tests/testServer.py \
-      --replace "def testBind(self):" "def dontTestBind(self):" \
-      --replace "def testBindv6(self):" "def dontTestBindv6(self):"
+      --replace-warn "def testBind(self):" "def dontTestBind(self):" \
+      --replace-warn "def testBindv6(self):" "def dontTestBindv6(self):" \
+
+    # A lot of test methods have been deprecated since Python 3.1
+    # and have been removed in Python 3.12.
+    # https://docs.python.org/3/whatsnew/3.11.html#pending-removal-in-python-3-12
+    substituteInPlace tests/*.py \
```

Comment:
we should create PR upstream

Reply:
Actually, this is my bad. Just re-checked upstream and saw that even though there was the release for the last 3 years, the package is still maintained, and this issue has been solved.
https://github.com/pyradius/pyrad/commit/0d5237d6df4d937aab63a82e0a7c4dc18e58ed88

Should I keep the package at the last release, or should I update it to a more recent rev?